job_creator.php

```diff
@@ -0,0 +1,270 @@
+                            'phpunit' => true,
+                            'phpunit_suite' => 'all'
+                        ]);
+                        $matrix['include'][] = $this->createJob(3, [
+                            'db' => DB_MYSQL_80,
+                            'phpunit' => true,
+                            'phpunit_suite' => 'all'
+                        ]);
+                    }
+                }
+            }
+            // skip phpcs on silverstripe-installer which include sample file for use in projects
+            if ((file_exists('phpcs.xml') || file_exists('phpcs.xml.dist')) && !preg_match('#/silverstripe-installer$#', $githubRepository)) {
+                $matrix['include'][] = $this->createJob(0, [
+                    'phplinting' => true
+                ]);
+            }
+            // phpcoverage also runs unit tests
+            // always run on silverstripe account
+            if ($run['phpcoverage'] || preg_match('#^silverstripe/#', $githubRepository)) {
+                if ($simpleMatrix) {
+                    $matrix['include'][] = $this->createJob(0, [
+                        'phpcoverage' => true
+                    ]);
+                } else {
+                    $matrix['include'][] = $this->createJob(2, [
+                        'db' => DB_MYSQL_57_PDO,
+                        'phpcoverage' => true
+                    ]);
+                }
+            }
+            // endtoend / behat
+            if ($run['endtoend'] && file_exists('behat.yml')) {
+                $matrix['include'][] = $this->createJob(0, [
+                    'endtoend' => true,
+                    'endtoend_suite' => 'root'
+                ]);
+                if (!$simpleMatrix) {
+                    $matrix['include'][] = $this->createJob(3, [
+                        'db' => DB_MYSQL_80,
+                        'endtoend' => true,
+                        'endtoend_suite' => 'root'
+                    ]);
+                }
+            }
+            // javascript tests
+            if (file_exists('package.json') && $run['js']) {
+                $matrix['include'][] = $this->createJob(0, [
+                    'js' => true
+                ]);
+            }
+        }
+
+        // extra jobs
+        foreach ($extraJobs as $arr) {
+            $matrix['include'][] = $this->createJob(0, $arr);
+        }
+
+        // convert everything to strings and sanatise values
+        foreach ($matrix['include'] as $i => $job) {
+            foreach ($job as $key => $val) {
+                if ($val === true) {
+                    $val = 'true';
+                }
+                if ($val === false) {
+                    $val = 'false';
+                }
+                // all values must be strings
+                $val = (string) $val;
+                // remove any dodgy characters
+                $val = str_replace(["\r", "\n", "\t", "'", '"', '&', '|'], '', $val);
```

Comment:
```suggestion
                $val = str_replace(["\f", "\r", "\n", "\t", "'", '"', '&', '|'], '', $val);
```
ref: https://www.asciihex.com/character/control/12/0x0C/ff-form-feed
Not a common character, but since we're not using regex `\v` we should include it here explicitly.
In fairness it is _probably_ covered by the next replacement anyway, but it pays to be sure.